### ISSUE TITLE
Measuring retriever performance for different K

### ISSUE THREAD

**Issue by geekydevu** [facebookresearch/DPR#129]
Hi, 

Thanks for sharing the code of your paper! Upon running train_dense_encoder.py, we get something like "correct prediction ratio  %d/%d ~  %f 0.6334815222466311 1177 1496 0.7867647058823529". Is this the top-100 accuracy? 

Could you also tell how to compute top-K accuracy in general via code for a given validation set?

Thanks in advance,
Devang.

**Comment by vlad-karpukhin** [comment]
correct prediction ratio is the ration of dev samples which are predicted correctly - i.e. positive passage has the highest similarity score among the candidates list.
Candidates list is the same as composed by in-batch negative scheme + hard negatives just with a different batch size from the train samples (see dev_batch_size param). 

So, this is NOT top-100 accuracy.

In order to get top-k accuracy you need to generate embeddings for the entire wikipedia first and then run dense_retriever tool. All these steps are described in the README.

**Comment by geekydevu** [comment]
Thanks, I was able to reproduce results successfully. Closing the issue.
